RestructuredText/reStructuredText.sublime-syntax

```diff
@@ -27,6 +27,17 @@ contexts:
           pop: true
         - meta_content_scope: text.html.basic
         - include: scope:text.html.basic
+    - match: '^(\s*)((\.\.)\s+math(::))\s*'
+      comment: directives.math
+      captures:
+        2: meta.directive.restructuredtext
+        3: punctuation.definition.directive.restructuredtext
+        4: punctuation.separator.key-value.restructuredtext
+      push:
+        - match: '^(?!\1\s+)(?=\s*\S+)'
+          pop: true
+        - meta_content_scope: text.tex.latex meta.environment.math.block.restructuredtext
+        - include: scope:text.tex.latex#math-content
```

Comment:
Have you tried using `embed` for this?

Reply:
What's the difference between this approach and that one?

Reply:
`embed` ensures that no matter what context the "embedded" syntax/context is in, it always pops all the way out as soon as the `escape` pattern is found.

For instance, if `math-content` pushed into a sub-context for a `(`, it may get stuck in there if there isn't a `)`. If using `embed` and `escape`, the first thing that happens is that we look for the `escape` pattern. If that is found, any content until then is parsed using the specified context. So it wouldn't matter if there is no `)`, it will exit the embedded context once finding the `escape` pattern.

Beyond that, the special nature of `embed` allows us to apply certain optimizations behind the scenes for performance. There will likely also be more optimizations made in the future.